### PR TITLE
Add natively instrumented to glossary

### LINKED ISSUE #4089: [Terminology] Add a term to the glossary that describes when "Instrumented Library_and Instrumentation Library are the same"
**What are you trying to achieve?**

To kick off the effort of what was described in https://github.com/open-telemetry/community/issues/2165 I'd like to request that we add a term to the specification glossary that describes the case where [_Instrumented Library_ and _Instrumentation Library_ may be the same library if it has built-in OpenTelemetry instrumentation.](https://opentelemetry.io/docs/specs/otel/glossary/#instrumentation-library)

A term often used for that is **native instrumentation**, but other terms [have been suggested](https://github.com/open-telemetry/opentelemetry.io/discussions/3808), since "[native](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/n/native-language)" might mean lots of things that are unrelated to technical aspects".

**Goal:** We need this term for the documentation to describe the process of adding OpenTelemetry to a library for out of the box telemetry. We also want to use it for the registry to tag entries of "native" instrumentations. So a term that is easy to understand even without the word "instrumentation" would be helpful (like 'native', 'out of the box', etc.), e.g. for the "Instrumentation for Next.js", see https://opentelemetry.io/ecosystem/registry/?s=next.js

The case where an application comes with OpenTelemetry out of the box (like some [K8s System Components](https://kubernetes.io/docs/concepts/cluster-administration/system-traces/)) needs also to be covered or explicitly excluded.

The term should also make it clear that it is the opposite of when instrumentation is added externally, e.g. by an instrumentation library.

cc @open-telemetry/docs-approvers @open-telemetry/technical-committee @open-telemetry/governance-committee

### PR DESCRIPTION
**Pull request by svrnm**
Fixes #4089

## Changes

Based on the issue discussion and the feedback from yesterdays' SIG call I propose a change for adding the term "natively instrumented" to the glossary. The definition especially uses existing wording (instrumenting library is instrumentation library) and some of the words by @tigrannajaryan in [this comment](https://github.com/open-telemetry/opentelemetry-specification/issues/4089#issuecomment-2286510672). I think it is important to have a sentence that also makes sure that there is a clear line between "natively instrumented" and things that are close enough (like a plugin that adds otel in a second step, even if it is native to the ecosystem) 

* [x] Related issues #4089
* [x] [`CHANGELOG.md`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/CHANGELOG.md) file updated for non-trivial changes

### CODE REVIEW

File: specification/glossary.md [hunk 1]
```diff
@@ -112,6 +113,16 @@ monkey patching, or running eBPF programs.
 
 Synonym: *Auto-instrumentation*.
 
+### Natively instrumented
+
+Denotes a library or application that has built-in instrumentation by using the OpenTelemetry API directly, thus
```
Comment (cijothomas) [inline specification/glossary.md:118]: One thing that would be good to clarify: 
If a library is instrumented natively using OTel API, but does not follow otel semantic conventions, is that still considered natively instrumented?

(In .NET world, many libraries part of ecosystem like HTTPClient, Asp.NET core uses Otel API and produce spans. But it does not populate attributes. So we currently have instrumentation libraries that act as a enricher, by populating attributes using otel sem. convention. Within .NET world,  the term "natively" instrumented is used to refer to those .NET libraries, even though they don't follow otel sem. convention.)

Comment (svrnm) [reply specification/glossary.md:118]: I would say yes, the moment an application makes use of the API, it's "natively instrumented", we can still take about "levels of instrumentation", i.e. do they use semantic conventions? do they use all signals? do they provide exemplars?, etc., but the purpose of this label is to establish what is the goal ("native instrumentation") and what is a gap solution ("instrumentation libraries")

File: specification/glossary.md [hunk 2]
```diff
@@ -112,6 +113,16 @@ monkey patching, or running eBPF programs.
...
+*Instrumented Library* and *Instrumentation Library* are the same library.
+
+If a separate *instrumentation library* or a plugin is required to instrument a library or application,
+that *instrumented library* or application MUST not be called *natively instrumented*, even if the
```
Comment (lmolkova) [inline specification/glossary.md:122]:
```suggestion
that *instrumented library* or application must not be called *natively instrumented*, even if the
```

We have no means to enforce this and having normative language might not be helpful - we don't use it for other terms in this document.

**Review by lmolkova — APPROVED** [review]

**Review by trask — APPROVED** [review]

**Review by tigrannajaryan — APPROVED** [review]

**Review by jack-berg — APPROVED** [review]

File: specification/glossary.md [hunk 3]
```diff
@@ -112,6 +113,16 @@ monkey patching, or running eBPF programs.
 
 Synonym: *Auto-instrumentation*.
 
+### Natively instrumented
+
+Denotes a library or application that has built-in instrumentation by using the OpenTelemetry API directly, thus
+*Instrumented Library* and *Instrumentation Library* are the same library.
```
Comment (MrAlias) [inline specification/glossary.md:119]: These are lower case below.

```suggestion
*instrumented library* and *instrumentation library* are the same library.
```

File: specification/glossary.md (hunk 1 above)
Comment (MrAlias) [inline specification/glossary.md:118]: "built-in" is an adjective, not a verb.

```suggestion
Denotes a library or application that has used the OpenTelemetry API directly to provide built-in instrumentation, thus
```

File: specification/glossary.md [hunk 4]
```diff
@@ -112,6 +113,16 @@ monkey patching, or running eBPF programs.
...
+
+If a separate *instrumentation library* or a plugin is required to instrument a library or application,
+that *instrumented library* or application must not be called *natively instrumented*, even if the
```
Comment (arminru) [inline specification/glossary.md:122]:
```suggestion
that *instrumented library* or application MUST NOT be called *natively instrumented*, even if the
```

Comment (cijothomas) [reply specification/glossary.md:122]: it was intentionally done this way per feedback earlier: https://github.com/open-telemetry/opentelemetry-specification/pull/4186#discussion_r1717696709

Comment (arminru) [reply specification/glossary.md:122]: I see, thanks!

File: specification/glossary.md [hunk 5]
```diff
@@ -21,6 +21,7 @@ Some other fundamental terms are documented in the [overview document](overview.
   * [In-band and Out-of-band Data](#in-band-and-out-of-band-data)
   * [Manual Instrumentation](#manual-instrumentation)
   * [Automatic Instrumentation](#automatic-instrumentation)
+  * [Natively instrumented](#natively-instrumented)
```
[the remark this answers is not in the document]
Comment (svrnm) [reply specification/glossary.md:24]: I am fine with both, but would prefer "Natively Instrumented" then

[the remark this answers is not in the document]
Comment (svrnm) [reply specification/glossary.md:24]:
```suggestion
  * [Natively Instrumented](#natively-instrumented)
```

File: specification/glossary.md [hunk 6]
```diff
@@ -112,6 +113,16 @@ monkey patching, or running eBPF programs.
 
 Synonym: *Auto-instrumentation*.
 
+### Natively instrumented
```
Comment (svrnm) [inline specification/glossary.md:116]:
```suggestion
### Natively Instrumented
```

**Review by reyang — APPROVED** [review]

**Pull request merged**